### PR TITLE
Add onClientObjectMoveStart, onClientObjectMoveStop, isObjectMoving

### LINKED ISSUE #874: isObjectMoving(), onObjectMoved
**Is your feature request related to a problem? Please describe.**
We can move an object using `moveObject()` function, but we cannot know whether the object has finished moving. 

**Describe the solution you'd like**
Add a new function `isObjectMoving()` which returns `true` if object moving or `false` otherwise. Add a new event `onObjectMoved`.

**Describe alternatives you've considered**
In some cases I can store target coords and then compare the current position with them, but it won't work with some types of easing (elastic, outback, bounce, etc).

**Additional context**
no

### PR DESCRIPTION
**Pull request by fresholia**
Fixes #874
Added 2 events 1 functions:
```lua
bool isObjectMoving(object)
event onClientObjectMoveStart
event onClientObjectMoveStop
```

Code on lua:
```lua
local obj = createObject(5239, -2417.22339, -606.70374, 132.56250)
moveObject(obj, 3000, -2417.22339, -606.70374, 137.56250)

addEventHandler("onClientObjectMoveStart", obj,
    function()

    end
)

addEventHandler("onClientObjectMoveStop", obj,
    function()
        outputChatBox("done")
    end
)
setTimer(
    function()
        outputChatBox(tostring(isObjectMoving(obj)))
    end,
500, 0)
```

### CODE REVIEW

File: Client/mods/deathmatch/logic/CDeathmatchObject.cpp [hunk 1]
```diff
@@ -96,6 +98,14 @@ void CDeathmatchObject::UpdateMovement()
 
     SetOrientation(positionRotation.m_vecPosition, positionRotation.m_vecRotation);
 
+    CLuaArguments Arguments;
+    CVector       currentPosition;
+    this->GetPosition(currentPosition);
+    Arguments.PushNumber(currentPosition.fX);
+    Arguments.PushNumber(currentPosition.fY);
+    Arguments.PushNumber(currentPosition.fZ);
+    this->CallEvent("onClientObjectMove", Arguments, true);
```
Comment (botder) [inline Client/mods/deathmatch/logic/CDeathmatchObject.cpp]: This is going to crash the client if somebody decides to use `stopObject` or `moveObject` inside that event handler.

**Review by Lpsd — CHANGES_REQUESTED** [review]
Small thing, single-line ifs like this:
```cpp
if (true)
    foo.DoSomething();
```

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 2]
```diff
@@ -3865,14 +3865,17 @@ bool CStaticFunctionDefinitions::IsObjectBreakable(CClientObject& Object, bool&
...
+
     if (IS_OBJECT(&Entity))
-    {
         bIsMoving = Object.IsMoving();
-    }
-    return true;
+    else
+        bIsMoving = false;
+
+    return bIsMoving;
```
Comment (Lpsd) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: Remove this and just initialize `bIsMoving` as false.

Comment (Lpsd) [reply Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: Didn't mean to include the `return bIsMoving;` line

**Review by Lpsd — COMMENTED** [review]
Looks good, left you one thing to deal with.

File: Client/mods/deathmatch/logic/CDeathmatchObject.cpp [hunk 3]
```diff
@@ -96,6 +100,7 @@ void CDeathmatchObject::UpdateMovement()
 
     SetOrientation(positionRotation.m_vecPosition, positionRotation.m_vecRotation);
 
+
```
Comment (Lpsd) [inline Client/mods/deathmatch/logic/CDeathmatchObject.cpp]: Unnecessary additional newline

**Review by Lpsd — CHANGES_REQUESTED** [review]
Good work, just fix that extra newline and we're good.

**Review by Lpsd — APPROVED** [review]

**Comment by patrikjuvonen** [comment]
@cleopatradev Please create new Wiki pages for these events and the function, thank you.
https://wiki.multitheftauto.com/wiki/Main_Page

**Pull request merged**